# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/TestingFunctions.cpp
# Commit: e39ae7aad8d9
# Full Hash: e39ae7aad8d93497b7a7e6ccf4ee675b4631c28e
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2023-09-21 03:21:32
# Regressor Bug: 1690111
# File Overlap Count: 2
# Description:
#   Bug 1690111 - Implement JS::EnsureNonInlineArrayBufferOrView. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D176596
# ==============================================================================

diff -r 603e8067e68f -r e39ae7aad8d9 js/src/builtin/TestingFunctions.cpp
--- a/js/src/builtin/TestingFunctions.cpp	Wed Sep 20 09:41:59 2023 +0000
+++ b/js/src/builtin/TestingFunctions.cpp	Wed Sep 20 09:42:00 2023 +0000
@@ -5687,6 +5687,28 @@
   return true;
 }
 
+static bool EnsureNonInline(JSContext* cx, unsigned argc, Value* vp) {
+  CallArgs args = CallArgsFromVp(argc, vp);
+
+  if (args.length() != 1) {
+    JS_ReportErrorASCII(cx, "ensureNonInline() requires a single argument");
+    return false;
+  }
+
+  if (!args[0].isObject()) {
+    JS_ReportErrorASCII(cx, "ensureNonInline must be passed an object");
+    return false;
+  }
+
+  RootedObject obj(cx, &args[0].toObject());
+  if (!JS::EnsureNonInlineArrayBufferOrView(cx, obj)) {
+    return false;
+  }
+
+  args.rval().setUndefined();
+  return true;
+}
+
 static bool JSONStringify(JSContext* cx, unsigned argc, Value* vp) {
   CallArgs args = CallArgsFromVp(argc, vp);
 
@@ -9498,6 +9520,11 @@
 "     2 - fail during read() hook\n"
 "  Set the log to null to clear it."),
 
+    JS_FN_HELP("ensureNonInline", EnsureNonInline, 1, 0,
+"ensureNonInline(view or buffer)",
+"  Ensure that the memory for the given ArrayBuffer or ArrayBufferView\n"
+"  is not inline."),
+
     JS_FN_HELP("JSONStringify", JSONStringify, 4, 0,
 "JSONStringify(value, behavior)",
 "  Same as JSON.stringify(value), but allows setting behavior:\n"